lib/elixir/pages/anti-patterns/code-anti-patterns.md

```diff
@@ -324,6 +324,13 @@ When a key is optional, the `map[:key]` notation must be used instead. This way,
 
 When you use `map[:key]` to access a key that always exists in the map, you are making the code less clear for developers and for the compiler, as they now need to work with the assumption the key may not be there. This mismatch may also make it harder to track certain bugs. If the key is unexpectedly missing, you will have a `nil` value propagate through the system, instead of raising on map access.
 
+*Table: Comparison of Map Access Notations*
+
+| Access Notation | Key Exists        | Key Doesn't Exist | Use Case      |
```

Comment:
This made me realize that there's actually an even bigger difference that isn't described in the text above either, which is `map` being `nil`.
So `map[:key]` would be "optional map, optional key".
We could have a `Map.get(map, :key)` row which is for "required map, optional key", (and maybe `Map.fetch!(map, "key")` row for non-atom required keys.

Reply:
And also pattern-matching which comes up later. So maybe it's best to add it at the end of the explanation as a form of recap?